finagle-redis/src/main/scala/com/twitter/finagle/Redis.scala

```diff
@@ -63,4 +66,48 @@ object Redis extends Client[Command, Reply] {
 
   def newService(dest: Name, label: String): Service[Command, Reply] =
     client.newService(dest, label)
+
+  object Subscribe extends com.twitter.finagle.Client[SubscribeCommand, Unit] {
+
+    object SubscribeClient {
+      /**
+       * Default stack parameters used for redis client.
+       */
+      val defaultParams: Stack.Params = StackClient.defaultParams +
+        param.ProtocolLibrary("redis.subscribe")
+
+      /**
+       * A default client stack which supports the pipelined redis client.
+       */
+      def newStack: Stack[ServiceFactory[SubscribeCommand, Unit]] = StackClient.newStack
+        .replace(DefaultPool.Role, SingletonPool.module[SubscribeCommand, Unit])
+    }
+
+    case class SubscribeClient(
```

Comment:
That's not the same. We may subscribe multiple times, but it will not change the way how we communicate with the server. In the subscribe mode, there's no correspondence between the outgoing and incoming messages, while in the ordinary mode, the outgoing and incoming messages are in one-by-one correspondence (so pipelining works). I think it's more appropriate to consider them two different protocols, despite that they encode messages in the same manner. How about we pull up Redis.Subscribe, rename it to RedisSubscribe, add some methods to create rich subscribe clients, and document the designing considerations in its scaladoc? We could also add a README for finagle-redis to provide some get-started examples.